sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidQuery.java

```diff
@@ -1382,17 +1382,13 @@ private WindowOperatorQuery toWindowQuery()
     if (windowing == null) {
       return null;
     }
-
-    final DataSource myDataSource;
     if (dataSource instanceof TableDataSource) {
-      // In this case, we first plan a scan query to pull the results up for us before applying the window
-      myDataSource = new QueryDataSource(toScanQuery());
-    } else {
-      myDataSource = dataSource;
+      // We need a scan query to pull the results up for us before applying the window
+      return null;
```

Comment:
no; this only means that this composition of the query is invalid - the planner should come up with a different configuration: `ScanQuery` and then a `WindowQuery` on top - instead of just pushing in the `window` as well into this `DruidQuery`